extensions/macos-utilities/src/utils.tsx

```diff
@@ -0,0 +1,76 @@
+import fs from "fs";
+import os from "os";
+import path from "path";
+import { execSync } from "child_process";
+
+export const backupLocation = "/tmp/terminal-history-backup.file.txt";
+
+/// Parse the shell history file
+/// - Parameter string: The string to parse
+/// - Returns: An array of commands
+export function parseShellHistory(string: string) {
+  const reBashHistory = /^: \d+:0;/;
+
+  return string
+    .trim()
+    .split("\n")
+    .map((line) => {
+      if (reBashHistory.test(line)) {
+        const lines = line.split(";").slice(1).join(";");
+        // Remove duplicates and return
+        return [...new Set(lines.split(":"))].join(":");
+      }
+
+      // ZSH just places one command on each line
+      return line;
+    });
+}
+
+/// Get the path to the shell history file
+/// - Returns: The path to the shell history file
+export function shellHistoryPath() {
+  const zshHistoryPath = path.join(os.homedir(), ".zsh_history");
+  const bashHistoryPath = path.join(os.homedir(), ".bash_history");
+
+  if (fs.existsSync(zshHistoryPath)) {
```

Comment:
The core of this plugin seems to all be synchronous. Given that history files could be significant in size, would it be worth using the async versions of all these functions and using `usePromise` at the command level?

Reply:
Thanks for the feedback here. Yes, the core was syncronous. I've moved this extension towards an asyncronous approach in https://github.com/raycast/extensions/pull/9981/commits/6142c027e0c299f47f6554f6615583060a1b8628